### ISSUE TITLE
`FileAccess` `get_path` contains an empty string when opened with compression.

### ISSUE THREAD

**Issue by asolux** [godotengine/godot#70712]
### Godot version

4.0.beta10

### System information

Windows 10, 64bit

### Issue description

The title says it all.
Opened in compression mode, whether read or write and whether the file already exists or not, `get_path` contains an empty string. This is also the case in C#.
However, if the file is opened without compression, get_path contains the absolute file path to the file as expected.
It doesn't matter whether an internal res:// path or an external system path is used.

### Steps to reproduce

```
var file = FileAccess.open_compressed("res://file.txt", FileAccess.WRITE")
prints("get_path():", file.get_path())
```
output: `get_path():`

```
var file = FileAccess.open("res://file.txt", FileAccess.WRITE)
prints("get_path():", file.get_path())
```
output: `get_path(): res://file.txt`

 
_or under C#:_
```
using var file = FileAccess.OpenCompressed("res://test.txt", FileAccess.ModeFlags.Write);
GD.PrintS("GetPath():", file.GetPath());
```
output: `GetPath():`

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
Can you reproduce this in 3.5.1?

**Comment by asolux** [comment]
Yes, this also happens with 3.5.1, just tested. But also on Windows 10.

```
var file = File.new()
file.open_compressed("res://test.txt", File.WRITE)
prints("get_path():", file.get_path())
```
output `get_path():`

```
var file = File.new()
file.open("res://test.txt", File.WRITE)
prints("get_path():", file.get_path())
```
output: `get_path(): res://test.txt`

**Comment by heppocogne** [comment]
Here's Godot4 `FileAccess::get_path()` implementation:
(We should use pure virtual function instead of returning empty string, isn't it?)
https://github.com/godotengine/godot/blob/55da3c1631906d7daabcc8bb5b43a74235567f39/core/io/file_access.h#L108

And this function is not overridden in `FileAccessCompressed`. I think that's why `open_compressed` always returns empty string.
